libs/remix-ui/solidity-compiler/src/lib/logic/compileTabLogic.ts

```diff
@@ -131,6 +131,11 @@ export class CompileTabLogic {
 
   runCompiler (externalCompType) {
     try {
+      const currentFile = this.api.currentFile
+      if (currentFile && (currentFile.endsWith('.ts') || currentFile.endsWith('.js') || currentFile.endsWith('.txt') || currentFile.endsWith('.log'))) {
```

Comment:
I think it is better to check files available for compilation like .sol, .vy etc as there are a great number of other extensions possible.

I can still compile the .json and .md file

Reply:
yes I know, i'll find an alternative